### PR TITLE
EC2 enhancement  create vpc end point

### PR DESCRIPTION
**Pull request by usmangani1**
Enhancement create vpc end point

https://github.com/localstack/localstack/issues/1450

### CODE REVIEW

**Comment by usmangani1** [comment]
> Enhancement create vpc end point
> 
> [localstack/localstack#1450](https://github.com/localstack/localstack/issues/1450)

@bblommers  Can you please review this and help me get it merged to master

File: moto/ec2/models.py [hunk 1]
```diff
@@ -2877,6 +2881,66 @@ def associate_vpc_cidr_block(
...
+                             tag_specifications=None,
+                             private_dns_enabled=None
+                            ):
+
+        vpc_endpoint_id = generate_vpc_end_point_id(vpc_id)
+
+        #validates if vpc is present or not.
+        self.get_vpc(vpc_id)
+
+        if type == "interface" or "Interface ":
```
Comment (bblommers) [inline moto/ec2/models.py:2904]: This is evaluated as ```(type == "interface") or ("Interface")```
The string `Interface` will always evaluate to True, regardless of the outcome of `type == interface` - so this statement doesn't do what you think it does
```suggestion
        if type and type.lower() == "interface":
```

File: tests/test_ec2/test_route_tables.py [hunk 2]
```diff
@@ -618,3 +618,39 @@ def test_describe_route_tables_with_nat_gateway():
     nat_gw_routes[0]["DestinationCidrBlock"].should.equal("0.0.0.0/0")
     nat_gw_routes[0]["NatGatewayId"].should.equal(nat_gw_id)
     nat_gw_routes[0]["State"].should.equal("active")
+
```
Comment (bblommers) [inline tests/test_ec2/test_route_tables.py:621]: Can you add some test cases for the different VPC types ('Interface'|'Gateway'|None).

Also add some asserts that either DNS entries are created, or routes - but not both.

**Comment by bblommers** [comment]
Fixes #2826

**Comment by 111andre111** [comment]
@bblommers Any idea of when will be the next release cycle, because this a quite huge effort here?

**Comment by bblommers** [comment]
@111andre111 Every PR creates a new dev-release, available here: https://pypi.org/project/moto/#history
If I'm not mistaken, this feature should be available in moto >= 1.3.15.dev776

**Comment by 111andre111** [comment]
Cool, perfect, that's good to know.
